### DISCUSSION TITLE
Multiple Filament Forms in separate Livewire components on the same page

### DISCUSSION THREAD

**Discussion by GabrielSieben** [filamentphp/filament#8937] [Help]
Hello,

I have a Filament v2 (Livewire v3 isn't quite compatible yet for an upgrade) application with a fairly tricky setup: It has a Filament form in a Livewire component, but that Form is duplicated in several different instances of the same Livewire component on the same page. This works almost just fine; except that my forms have some components like radio buttons affect the other forms. 

For an example to easily reproduce this, imagine if you had a form called `SimpleForm`, with just a radio button selection between `Yes`, `Maybe`, and `No`. And you called `@livewire('simple-form')` twice instead of once. You would notice that you can't have the two forms having independent values - clicking `Yes` in one deselects all of the radio options in the separate component.

Are there any ideas about what can be done to fix this overlap? I've made sure to use `wire:key` but I'm very puzzled. 

Edit: Also, though I am not at liberty to publicly discuss the details, I do sincerely believe that if I had to privately explain why this was necessary, it would be quickly accepted that this is the only reasonable way to implement the features I am developing. 

Edit 2: After taking a closer look again, it's because the `name` attribute on the actual, final `<input>` tag is the same between forms. Could there maybe be a patch, or a workaround, to insert the parent `wire:key` appended onto the current `name` property to solve this?

**Comment by leandrocfe** [comment]
[using-multiple-forms](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#using-multiple-forms)?

**Reply by GabrielSieben** [reply]
I've seen that, that isn't it. That is very useful for putting multiple Filament Forms inside of a single component - so like it says there, I could have a 'editPostForm' and 'createCommentForm' inside a hypothetical `ResponseForm` Livewire component. I need the ability to have both of those forms be in separate Livewire components and not conflict with each other.

The main reason it won't work for me is because the amount of forms displayed is programatic, which gets extremely unwieldy if not impossible inside of one Livewire component. I also (if you can believe it), have different types of forms being programmatically displayed. (Think a page with 5 forms of kind X, 3 forms of kind Y, and 7 forms of kind Z; while a different page might have 2 forms of kind X, 6 forms of kind Y, and 3 forms of kind Z.)

**Comment by edwink75** [comment]
Did you find a solution for your problem?
I have a similar issue where the second for in in a popup (Custom with separate Livewire component, not Filament) and the whole form can not be edited. The forms have completely different field and I gave the forms different `getFormStatePath()`
Something is conflicting, but I don't get an error, just the second field set that is not accessible

**Reply by leandrocfe** [reply]
Could you share some code you are trying to do please?

**Comment by GabrielSieben** [comment]
The solution was to rewrite the quiz engine to have only one question on screen at a time. Sadly I could not fix the underlying issue.

Gabriel

Sent from my iPhone

On Feb 27, 2024, at 7:55 AM, Edwin Krause ***@***.***> wrote:

﻿

Did you find a solution for your problem?
I have a similar issue where the second for in in a popup (Custom with separate Livewire component, not Filament) and the whole form can not be edited. The forms have completely different field and I gave the forms different getFormStatePath()
Something is conflicting, but I don't get an error, just the second field set that is not accessible

**Comment by edwink75** [comment]
> Could you share some code you are trying to do please?

It's a bit hard to do so, but here is what I'm doing:
1. First form is loaded with Select, TextInput and TextArea fields in popup number one.
2. Click Button on first Form that opens a new Livewire component in an overlying popup with a second form.
3. The second form is not editable, I'm not even able to get the cursor into the fields

What have I tried:
1. different formStatePath for each form
2. different formContext (not sure what that does to be honest)
3. different form name attributes
4. I removed HasForm and InteractsWithForms from second Livewire component and added html code for a simple input type text field, which then also was unresponsive / inaccessible. 
5. Give up and search for help online :-)

Same as the author, we're using Filament v2 at this point. As the project is very complex we're planning to upgrade to v3 near the end of this year.
